src/main/java/baritone/process/MineProcess.java

```diff
@@ -420,6 +422,22 @@ private static List<BlockPos> prune(CalculationContext ctx, List<BlockPos> locs2
                     }
                 })
 
+                .filter(pos -> {
+                    if (Baritone.settings().onlyMineInSelection.value || Baritone.settings().onlyMineOutOfSelection.value) {
+                        boolean isInSelection = Arrays.stream(ctx.baritone.getSelectionManager().getSelections()).anyMatch(selection -> selection.aabb().contains(new Vec3d(pos.getX(), pos.getY(), pos.getZ())));
```

Comment:
This will potentially run many times per tick, not sure how much we can or should try to optimize it.

Reply:
From my tests this running multiple times a tick doesn't really matter as it is quite quick, it would slow the game down slightly, however I can't see an easy optimisation which would be worth the effort.